src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -31,7 +25,55 @@
     private GameDao gameDao;
     private GameUtil gameUtil;
 
+    @Override
+    public void endTurn(UserBean user, String gameIdString) throws PlayException, GameException {
+        log.debug("User {} tries to end his turn of game id {}",
+                user == null ? "<EMPTY>" : user.getUsername(), gameIdString);
+        //TODO: move to common validation method
+        if (user == null) {
+            log.debug("User should not be empty");
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+        GameBean game = gameUtil.getGameById(gameIdString, ERROR_CODE_ERROR);
+        if (game.getStatus() != GameStatus.PLAYING) {
+            log.debug("Cannot end turn of not playing game");
+            throw new GameException(ERROR_CODE_ERROR);
+        }
+
+        //TODO: move to util method and refactor all other places
+        GameUserBean gameUserBean = null;
+        for (GameUserBean gameUser : game.getGameUsers()) {
+            if (gameUser.getUser().equals(user)) {
+                gameUserBean = gameUser;
+                break;
+            }
+        }
+
+        if (gameUserBean == null) {
+            log.debug("User is not joined to game with specified id %s", gameIdString);
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+        if (!game.getCurrentMove().equals(gameUserBean.getMoveOrder())) {
+            log.debug("It is not current turn of user {}", user.getUsername());
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
 
+        //TODO: think about naming of method and namings of fields related to moveOrder and turn
```

Comment:
pass usually means 'peredat', while miss is used for 'propustit'. But naming is not critical so never mind